tiny_dnn/core/kernels/conv2d_op_avx.h

```diff
@@ -375,6 +390,7 @@ void avx_conv2d_5x5_kernel(const core::conv_params &params,
         size_t cnt  = (out_area - headSize) / 8;
         double *pa2 = pa + headSize;
         for (size_t i = 0; i < cnt; ++i) {
+          // TODO: segfault here even though adresses are ok
           _mm256_store_pd(&pa2[i * 8 + 0], b2);
```

Comment:
In `conv2d_op_avx.h` file, replace `_mm256_store_ps` to `_mm256_storeu_ps` and you also need to change `_mm256_store_pd` to `_mm256_storeu_pd` in the same file. I locally confirmed these changes will fix the segfault with `tiny_dnn_test` program.

It can be that `xtensor` library doesn't allocate memory with enough alignment (32bytes for AVX, ideally it should be 64bytes though).